libraries/botbuilder-core/src/configurationBotFrameworkAuthentication.ts

```diff
@@ -0,0 +1,222 @@
+
+    /**
+     * (Optional) The OpenID metadata document used for authenticating tokens coming from the channel. The "ToBotFromChannelOpenIdMetadataUrl" member takes precedence over this value.
+     */
+    [AuthenticationConstants.BotOpenIdMetadataKey]: t.String.optional(),
+
+    /**
+     * A string used to indicate if which cloud the bot is operating in (e.g. Public Azure or US Government).
+     *
+     * @remarks
+     * A `null` or `''` value indicates Public Azure, whereas [GovernmentConstants.ChannelService](xref:botframework-connector.GovernmentConstants.ChannelService) indicates the bot is operating in the US Government cloud.
+     *
+     * Other values result in a custom authentication configuration derived from the values passed in on the [ConfigurationBotFrameworkAuthenticationOptions](xef:botbuilder-core.ConfigurationBotFrameworkAuthenticationOptions) instance.
+     */
+    [AuthenticationConstants.ChannelService]: t.String,
+
+    /**
+     * Flag indicating whether or not to validate the address.
+     */
+    ValidateAuthority: t.String.Or(t.Boolean),
+
+    /**
+     * The Login URL used to specify the tenant from which the bot should obtain access tokens from.
+     */
+    ToChannelFromBotLoginUrl: t.String,
+
+    /**
+     * The Oauth scope to request.
+     *
+     * @remarks
+     * This value is used when fetching a token to indicate the ultimate recipient or `audience` of an activity sent using these credentials.
+     */
+    ToChannelFromBotOAuthScope: t.String,
+
+    /**
+     * The Token issuer for signed requests to the channel.
+     */
+    ToBotFromChannelTokenIssuer: t.String,
+
+    /**
+     * The OAuth URL used to get a token from OAuthApiClient.
+     */
+    OAuthUrl: t.String,
+
+    /**
+     * The OpenID metadata document used for authenticating tokens coming from the channel.
+     */
+    ToBotFromChannelOpenIdMetadataUrl: t.String,
+
+    /**
+     * The The OpenID metadata document used for authenticating tokens coming from the Emulator.
+     */
+    ToBotFromEmulatorOpenIdMetadataUrl: t.String,
+
+    /**
+     * A value for the CallerId.
+     */
+    CallerId: t.String,
+});
+
+/**
+ * Contains settings used to configure a [ConfigurationBotFrameworkAuthentication](xref:botbuilder-core.ConfigurationBotFrameworkAuthentication) instance.
+ */
+export type ConfigurationBotFrameworkAuthenticationOptions = t.Static<typeof TypedOptions>;
+
+const isRuntimeConfiguration = (val: unknown): val is Configuration => {
+    const config = val as Configuration;
+    return tests.isFunc(config.get) && tests.isFunc(config.set);
+};
+
+/**
+ * Creates a [BotFrameworkAuthentication](xref:botframework-connector.BotFrameworkAuthentication) instance from an object with the authentication values or a [Configuration](xref:botbuilder-dialogs-adaptive-runtime-core.Configuration) instance.
+ */
+export class ConfigurationBotFrameworkAuthentication extends BotFrameworkAuthentication {
```

Comment:
I was testing out a refactor that (I think) greatly simplifies this style of "configuration driven" types. [See here](https://github.com/microsoft/botbuilder-js/blob/jpg/cloud-adapter/libraries/botbuilder-core/src/configurationBotFrameworkAuthentication.ts#L21).

It's basically just a function that creates an instance of a particular type. This is probably more natural in JS (compared with .NET), but I think it's a lot simpler than constructing an `inner` instance and re-wrapping all the methods.

Reply:
Did a simliar thing to the [service client credential factory](https://github.com/microsoft/botbuilder-js/blob/jpg/cloud-adapter/libraries/botbuilder-core/src/configurationServiceClientCredentialFactory.ts#L14).

Reply:
Along with that, I simplified the configuration options a fair amount. I understand the purpose of the .NET constructor overrides and the desire to provide super flexible types, but I think I'm in favor of gradual additive complexity as the need arises. Specifically, I dropped the non-`Configuration` driven overrides. In practice, these types will likely only be used by the runtime code. With that in mind, it makes more sense (to me) to focus on `Configuration` parameterization rather.